docs/dapp/opl/README.md

```diff
@@ -5,19 +5,147 @@ empowers developers to add new functionality to smart contracts on the most
 popular EVM networks like Ethereum, BNB Chain, and Polygon with encrypted
 transactions and confidential state.
 
-## DApp Development
-
-The Oasis Privacy Layer contains a set of [smart contracts](https://github.com/oasisprotocol/sapphire-paratime/tree/main/contracts/contracts/opl)
-that integrate [Sapphire](../sapphire/README.mdx) into your existing and future
-Web3 applications, introducing confidentiality.
-
-Application developers on any EVM-compatible network can add confidential state
-and selective disclosures to their application through the usage of a smart
-contract deployed on OPL without migrating their application stack. OPL enabled
-smart contracts provide a default bridge for developers to pass messages
-between Sapphire and other networks.
+By using [a Solidity library](https://github.com/oasisprotocol/sapphire-paratime/tree/main/contracts/contracts/opl)
+that integrates [Sapphire](https://oasisprotocol.org/sapphire) into your
+existing and future Web3 applications, developers on any supported network can
+seamlessly add confidential state and selective disclosures to their dApps when
+using contract a deployed on Sapphire without leaving their existing networks.
 
 For more information about OPL and to catch the latest news, please visit the
-official [page](https://oasisprotocol.org/opl).
+[official OPL page](https://oasisprotocol.org/opl).
 
 ![Oasis Privacy Layer diagram](../images/opl/privacy-layer-diagram.png)
+
+The user submits a transaction on the Home network to a contract which uses
+`postMessage` to notify the <abbr title="State Guardian Network">SGN</abbr>
+that it should approve the cross-chain message. The Executor waits, when the SGN
+approves the message the Executor submits a transaction to the target contract
+on Sapphire.
+
+```mermaid
+sequenceDiagram
+    User ->> Home Contract: Transaction
+    Home Contract -->> SGN: SGN Watches Events
+    SGN -->> Executor: Waits for SGN Approval
+    Executor ->> Sapphire Contract: Submit Proof Tx
+```
+
+The Home Contract pays the SGN to watch and approve the message, but the
+Executor needs to be run by somebody willing to pay for the gas to submit
+transactions to the destination chain.
+
+## Quickstart
+
+A of pair contracts are linked bidirectionally 1-1 to each other across chains,
+with one end on Sapphire and the other on a supported EVM compatible chain (the
+Home Network). They can post and receive messages to & from each other using the
+message-passing bridge, but must register endpoints to define which messages
+they handle from each other.
+
+Start by adding the [`@oasisprotocol/sapphire-contracts`](http://npmjs.com/package/@oasisprotocol/sapphire-contracts) NPM package to your Hardhat or Truffle project so you can import `OPL.sol`:
```

Comment:
Reference `@oasisprotocol/sapphire-contracts` link below to avoid long lines here.


```
[`@oasisprotocol/sapphire-contracts`]: http://npmjs.com/package/@oasisprotocol/sapphire-contracts
```